plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/AIAssistant.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.model.ai;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.model.ai.completion.*;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+
+import java.util.concurrent.Flow;
+
+/**
+ * AI Assistant interface. Provides methods for AI-based operations.
+ */
+public interface AIAssistant {
```

Comment:
Why do we need a copy of all methods with/without `DAICompletionEngine engine` param? may be make it nullable>

Reply:
Classical users do not require multiple active engines. However, we want the option to specify an engine for testing. We can make the parameter nullable, but then we must explicitly pass null in our application code. Essentially, we have two versions of the method: one for application code and one for testing.